src/utils.rs

```diff
@@ -233,19 +236,51 @@ pub enum AddType {
     Ignored,
 }
 
+/// If added or updated, moves the file to the `dest`.
 pub async fn maybe_upgrade_xdc(
     app_info: &mut AppInfo,
     conn: &mut SqliteConnection,
+    _dest: &Path,
 ) -> anyhow::Result<AddType> {
-    Ok(
-        if db::app_version_exists(conn, &app_info.app_id, app_info.version).await? {
-            AddType::Ignored
-        } else if db::app_exists(conn, &app_info.app_id).await? {
-            db::create_app_info(conn, app_info).await?;
-            AddType::Updated
-        } else {
+    let add_type = if db::app_version_exists(conn, &app_info.app_id, app_info.version).await? {
+        AddType::Ignored
+    } else if db::app_exists(conn, &app_info.app_id).await? {
+        AddType::Updated
+    } else {
+        AddType::Added
+    };
+
+    match add_type {
+        AddType::Added | AddType::Updated => {
+            #[cfg(not(test))]
+            {
+                fs::copy(
+                    &app_info.xdc_blob_path,
+                    &_dest.join(
+                        app_info
+                            .xdc_blob_path
+                            .file_name()
+                            .context("Can't get file name from xdc_blob_dir")?,
+                    ),
+                )
+                .await
+                .with_context(|| {
+                    format!(
+                        "Failed to copy {} to {}",
+                        app_info.xdc_blob_path.display(),
+                        _dest.display()
+                    )
+                })?;
+                app_info.xdc_blob_path = _dest.join(
+                    app_info
+                        .xdc_blob_path
+                        .file_name()
+                        .context("Can't get file name from xdc_blob_dir")?,
+                );
```

Comment:
Made this conditional because for testing it makes no sense to actually move stuff around (breaks tests aswell). 
We could make it a "no-op" by copying it to the same location where the file came from but this seems better. 
Of cause, this code could also be moved to the two call origins but would then produce code duplication.